File: App/src/uk/co/armedpineapple/innoextract/IExtractService.java
package uk.co.armedpineapple.innoextract;

interface IExtractService {

    void extract(final String toExtract, final String extractDir,
            final ExtractCallback callback);

    void check(final String toExtract, final CheckCallback callback);

    interface ExtractCallback {
        void onProgress(int value, int max);

        void onSuccess();

        void onFailure(Exception e);
    }

    interface CheckCallback {
        void onResult(boolean valid);
    }

}

File: App/src/uk/co/armedpineapple/innoextract/IExtractService.kt
package uk.co.armedpineapple.innoextract

import java.io.*

internal interface IExtractService {

    fun extract(toExtract: File, extractDir: File,
                callback: ExtractCallback)

    fun check(toExtract: File, callback: CheckCallback)

    interface ExtractCallback {
        fun onProgress(value: Int, max: Int)

        fun onSuccess()

        fun onFailure(e: Exception)
    }

    interface CheckCallback {
        fun onResult(valid: Boolean)
    }

}
